TrackletAlgorithm/MatchEngineUnit.h

```diff
@@ -372,34 +460,54 @@ inline void advance() {
  ap_uint<kNBits_MemAddr> projseqs_[1<<MatchEngineUnitBase<VMProjType>::kNBitsBuffer];
 
  //Current projection
- ap_uint<4> nstubsall_[4];
- NSTUBS nstubs_, istub_;
- ap_uint<4> stubmask_;
+ ap_uint<kNBits_MemAddrBinned> nstubsall_[4];
+ NSTUB istub_;
+ NSTUBS nstubs_;
+ ap_uint<kNBits_MemAddrBinned> stubmask_;
  ap_uint<1> second_;
  ap_uint<1> phiPlus_;
- typename ProjectionRouterBuffer<BARREL, AllProjectionType>::PHIPROJBIN phiProjBin_;
+ ap_int<2> shift_;
  bool idle_;
  ap_uint<VMStubMECMBase<VMSMEType>::kVMSMEFinePhiSize> iphi_;
  BXType bx_;
  bool empty_;
  VMStubMECM<VMSMEType> stubdata__, stubdata__t, stubdata___; 
- typename ProjectionRouterBuffer<BARREL, AllProjectionType>::TCID tcid_;
- ProjectionRouterBuffer<BARREL, AllProjectionType> projbuffer_;
+ typename ProjectionRouterBuffer<VMProjType, AllProjectionType>::TCID tcid_;
+ ProjectionRouterBuffer<VMProjType, AllProjectionType> projbuffer_;
  ap_uint<kNBits_MemAddr> projseq_;
  bool isPSseed_;
 
+ NSTUB iuse_;
+ static constexpr int kNuse = 4; // Only 4 options
+ int nuse_;
+ ap_uint<2> use_[kNuse];
+
 
- typename ProjectionRouterBuffer<BARREL, AllProjectionType>::VMPZBINNOFLAG zbin;
+ typename ProjectionRouterBuffer<VMProjType, AllProjectionType>::VMPZBINNOFLAG zbin_, zbin__, zbin___, zbin__t;
```

Comment:
We should rename so that we don't have the __t variables, but just __, ___, and ____